torch/csrc/lazy/core/shape_inference.h

```diff
@@ -28,6 +28,12 @@ TORCH_API std::vector<torch::lazy::Shape> compute_shape_bernoulli(const at::Tens
 TORCH_API std::vector<torch::lazy::Shape> compute_shape_bernoulli(const at::Tensor & self, double p, c10::optional<at::Generator> generator);
 TORCH_API std::vector<torch::lazy::Shape> compute_shape_binary_cross_entropy(const at::Tensor & self, const at::Tensor & target, const c10::optional<at::Tensor> & weight, int64_t reduction);
 TORCH_API std::vector<torch::lazy::Shape> compute_shape_binary_cross_entropy_backward(const at::Tensor & grad_output, const at::Tensor & self, const at::Tensor & target, const c10::optional<at::Tensor> & weight, int64_t reduction);
+TORCH_API std::vector<torch::lazy::Shape> compute_shape_bitwise_and(const at::Tensor & self, const at::Scalar & other);
+TORCH_API std::vector<torch::lazy::Shape> compute_shape_bitwise_and(const at::Tensor & self, const at::Tensor & other);
+TORCH_API std::vector<torch::lazy::Shape> compute_shape_bitwise_or(const at::Tensor & self, const at::Scalar & other);
+TORCH_API std::vector<torch::lazy::Shape> compute_shape_bitwise_or(const at::Tensor & self, const at::Tensor & other);
```

Comment:
It looks like that the TorchScript backend has this op implemented but without the needs of a shape inference function, which indicates that the meta tensor has handled this well. https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/ts_native_functions.yaml#L26. Can you share why XLA needs this?

Reply:
When building XLA without these methods I get the following error:
```
Please add declare this function in /workspace/pytorch/torch/csrc/lazy/core/shape_inference.h:

and implement it in the the corresponding shape_inference.cpp file.

TORCH_API std::vector<torch::lazy::Shape> compute_shape_bitwise_and(const at::Tensor & self, const at::Scalar & other);
TORCH_API std::vector<torch::lazy::Shape> compute_shape_bitwise_or(const at::Tensor & self, const at::Scalar & other);
TORCH_API std::vector<torch::lazy::Shape> compute_shape_bitwise_xor(const at::Tensor & self, const at::Scalar & other);
```

The methods for scalar output are required. I can remove the methods for tensor output, though.

Reply:
yea, it would be better to remove the tensor ones.  We want to reuse the meta kernels where they exist.

Reply:
Done. There are now only methods accepting `(const at::Tensor & self, const at::Scalar & other)` for bitwise `and`, `or`, and `xor`.

Reply:
I was thinking maybe XLA doesn't correctly implement the Scalar version of the ops so they don't get decomposed into the tensor ones. Need to double check.

Reply:
@steventk-g Can you try only add `bitwise_and.Tensor`, `bitwise_or.Tensor`, and `bitwise_xor.Tensor` in the full_codegen section and see if the .Scalar counterparts work as well. Ops that have both a tensor and a scalar version will often have the scalar version turned into the tensor version in the dispatcher. Therefore, usually you don't need to code-gen the scalar ones. However, it's not guaranteed given there are too many ops in PyTorch.